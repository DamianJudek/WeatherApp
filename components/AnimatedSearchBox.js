import React from 'react';
import {
  Animated,
  Image,
  TouchableHighlight,
  StyleSheet,
  TextInput,
} from 'react-native';
import PropTypes from 'prop-types';
import searchIcon from '../assets/icons/search.png';

const styles = StyleSheet.create({
  icon: {
    height: 20,
    width: 20,
  },
  iconContainer: {
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 5,
    backgroundColor: 'transparent',
  },
  searchBox: {
    backgroundColor: 'rgba(100, 100, 100, 0.3)',
    height: 50,
  },
  inputActive: {
    width: '100%',
    textAlign: 'center',
    fontSize: 16,
    color: '#fff',
  },
  inputInactive: { width: 0, padding: 0 },
});

class AnimatedSearchBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      animation: new Animated.Value(0),

      textInput: React.createRef(),

      inputVisible: false,

    };
  }

  static propTypes = {
    handleCityInput: PropTypes.func,
  };

  focusTextInput = () => {
    this.state.textInput.current.focus();
  };

  blurTextInput = () => {
    this.state.textInput.current.blur();
  };

  onPress = () => {

 

    if (this.state.inputVisible === true) {
     this.blurTextInput();
      Animated.timing(this.state.animation, {
        toValue: 0,
        duration: 800,
        useNativeDriver: false,
      }).start();
      this.setState({ inputVisible: false });
    } else  {
      Animated.timing(this.state.animation, {
        toValue: 1,
        duration: 800,
        useNativeDriver: false,
      }).start();

      this.focusTextInput();

      this.setState({ inputVisible: true });

    }
  };

  render() {
    return (
      <>
        <TouchableHighlight
          onPress={this.onPress}
          underlayColor="rgba(95, 158, 160, 0.3)"
          style={styles.iconContainer}>
          <Image style={styles.icon} resizeMode="contain" source={searchIcon} />
        </TouchableHighlight>
        <Animated.View
          style={{
            flex: this.state.animation,
            ...styles.searchBox,
          }}>
          <TextInput
            onSubmitEditing={(e) => {
              this.props.handleCityInput(e);
              this.onPress();
            }}
            ref={this.state.textInput}
            autoCorrect={false}
            placeholderTextColor="#fff"
            style={

              this.state.active ? styles.inputActive : styles.inputInactive

            }
            placeholder={this.state.inputVisible ? 'Miasto' : ''}
          />
        </Animated.View>
      </>
    );
  }
}

export default AnimatedSearchBox;
