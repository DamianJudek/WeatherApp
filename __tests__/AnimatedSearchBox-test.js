import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AnimatedSearchBox from '../components/AnimatedSearchBox';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

test('renders correctly', () => {
  const snap = renderer.create(<AnimatedSearchBox />).toJSON();
  expect(snap).toMatchSnapshot();
});
