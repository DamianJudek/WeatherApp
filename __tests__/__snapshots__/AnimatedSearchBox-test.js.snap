// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/icons/search.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "rgba(100, 100, 100, 0.3)",
        "flex": 0,
        "height": 50,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onSubmitEditing={[Function]}
      placeholder=""
      placeholderTextColor="#fff"
      rejectResponderTermination={true}
      style={
        Object {
          "padding": 0,
          "width": 0,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>,
]
`;
